### PR TITLE
Search: Support substring searches in the new search engine

### PR DESCRIPTION
**Pull request by ArturWierzbicki**
https://raintank-corp.slack.com/archives/C03FDAJ4VJT/p1661885430662479

### CODE REVIEW

File: pkg/services/searchV2/bluge.go [hunk 1]
```diff
@@ -444,6 +444,11 @@ func doSearchQuery(
 				SetOperator(bluge.MatchQueryOperatorAnd). // all terms must match
 				SetAnalyzer(ngramQueryAnalyzer).SetBoost(1))
 
+		bq.AddShould(
+			bluge.NewWildcardQuery(fmt.Sprintf("*%s*", q.Query)).
```
Comment (ArturWierzbicki) [inline pkg/services/searchV2/bluge.go:448]: I assume this can get slow - still looking for alternatives

Comment (ArturWierzbicki) [reply pkg/services/searchV2/bluge.go:448]: I guess this should be solved by the ngram analyzer, but we are currently using just the [punctuation chars](https://github.com/grafana/grafana/blob/0975ea4df807df6c96a73731c2c00e05bdfd1b62/pkg/services/searchV2/ngram.go#L14) and whitespaces to split input into tokens

Comment (FZambia) [reply pkg/services/searchV2/bluge.go:448]: Possibly sth like: https://github.com/grafana/grafana/compare/substring-searches...name_contains_query, seems order of magnitude faster, though I may have missed sth..

(note: it results into re-ordering of items in ordering benchmarks)

Comment (FZambia) [reply pkg/services/searchV2/bluge.go:448]: Ah, skip - missed Artur's bench branch

**Pull request closed without merging**